### ISSUE TITLE
Error: "unsupported feature: unary operator: !" in IF in trigger definition

### ISSUE THREAD

**Issue by florianlaws** [dolthub/dolt#5470]
When restoring a mysqldump file that also contains trigger definitions, I encountered the following error:
```
error on line 11823 for query /*!50003 CREATE*/ /*!50017 DEFINER=`root`@`localhost`*/ /*!50003 TRIGGER `before_task_status_column_update` BEFORE UPDATE ON `task` FOR EACH ROW BEGIN
  DECLARE OLD_STATUS_CATEGORY VARCHAR(191);
  DECLARE NEW_STATUS_CATEGORY VARCHAR(191);
  IF !(NEW.status_column_id <=> OLD.status_column_id) THEN
    SELECT category into OLD_STATUS_CATEGORY from forecast_db.status_column where id = OLD.status_column_id;
	SELECT category into NEW_STATUS_CATEGORY from forecast_db.status_column where id = NEW.status_column_id;
    IF (NEW_STATUS_CATEGORY = "DONE" AND OLD_STATUS_CATEGORY != "DONE") THEN
		SET NEW.set_done_at = CURRENT_TIMESTAMP;
    ELSEIF (NEW_STATUS_CATEGORY != "DONE" AND OLD_STATUS_CATEGORY = "DONE") THEN
		SET NEW.set_done_at = null;
    END IF;
  END IF;
END */: unsupported feature: unary operator: !
unsupported feature: unary operator: !
```

Changing `IF !(NEW.status_column_id ... ` to `(IF NOT(NEW.status_column_id ... ` works.

According to the [documentation](https://docs.dolthub.com/sql-reference/sql-support/expressions-functions-operators) , `!` should be supported as well though?

**Comment by timsehn** [comment]
We had a couple other paying customer bugs come in this morning so this one might not get a prompt a fix as your last couple.

**Comment by jennifersp** [comment]
Hi @florianlaws, thank you for reporting this issue. The fix for this issue has been merged into `go-mysql-server` and will be in the next Dolt release.

**Comment by florianlaws** [comment]
I can confirm the fix works in `0.54.2`. I'm amazed by your quick fixes. Thanks a lot y'all!

**Comment by timsehn** [comment]
Keep the issues coming. These "first hour" compatibility issues get a lot of attention from us. We know importing an existing MySQL database just has to work. If you have feature requests, those might not get done as fast :-) But buyt a support contract and we'll grind like this on anything you want :-)
